### ISSUE TITLE
Linear Regression의 cost 최소화 알고리즘 (lec3)

### ISSUE THREAD

**Issue by sculove** [studye/yeahEE#4]
### Gradient descent algorithm
- 경사를 따라 내려가는 알고리즘.
- 경사도를 따라서 한 발자국 움직이는 알고리즘
- W값을 바꿔가면서 반복해간다.
- 경사도? === 미분으로 구한다. `시팔`

#### 공식
<img width="372" alt="2017-10-07 11 16 11" src="https://user-images.githubusercontent.com/1383952/31308698-fd272bc4-abb5-11e7-8636-06c73cad30c1.png">
<img width="478" alt="2017-10-07 9 40 19" src="https://user-images.githubusercontent.com/1383952/31307933-253fe7c6-aba8-11e7-8dfc-d13451eced74.png">

- 미분 계산을 쉽게 하기 위해서 기존 cost 함수에서 1/2를 곱함.
- a : learning rate (상수)
- cost값을 미분함.
- 그값을 원래 W값에서 뺀다. 

> ##### 알아봐야할 수학 용어
> - 수렴과 발산 : 극한 값이 존재하는 경우 수렴. 그렇지 않은 경우 발산.
> - lim
> - 미분
>    - 미분은 함수의 `도함수`를 구하는 것. 
>    - 도함수는 주어진 함수의 `미분계수`를 함수값으로 가지는 함수.
>    - 미분계수는 `순간변화율`인 동시에 `접선의 기울기`
> http://unolab.tistory.com/entry/%EB%AF%B8%EB%B6%84%EC%9D%98-%EC%A0%95%EC%9D%98

### 문제점.
<img width="449" alt="2017-10-07 10 05 47" src="https://user-images.githubusercontent.com/1383952/31308111-b4b86b8c-abab-11e7-98f9-d566305997d6.png">

- 초기 시작점에 따라 다른 길로 가게되어 최소값이 잘못 찾을수 있다.
- convex function: 밥그릇을 엎어놓은 듯한 함수
- 따라서,  cost function의 모양이 반드시 `convex function`이 되는지 확인해야한다!!!
